core/src/main/java/com/linecorp/armeria/client/retry/RetryingHttpClient.java

```diff
@@ -135,19 +135,17 @@ private void doExecute0(ClientRequestContext ctx, HttpRequestDuplicator rootReqD
 
         final HttpResponseDuplicator resDuplicator =
                 new HttpResponseDuplicator(response, maxSignalLength(ctx.maxResponseLength()), ctx.eventLoop());
-        final ContentPreviewResponse contentPreviewResponse =
-                new ContentPreviewResponse(resDuplicator.duplicateStream(), contentPreviewLength);
-        retryStrategy().shouldRetry(rootReqDuplicator.duplicateStream(), contentPreviewResponse)
-                       .handle(voidFunction((backoffOpt, unused) -> {
-                           if (backoffOpt.isPresent()) {
+        retryStrategy().shouldRetry(rootReqDuplicator.duplicateStream(), contentPreviewResponse(resDuplicator))
+                       .handle(voidFunction((backoff, unused) -> {
+                           if (backoff != null) {
                                final long millisAfter =
-                                       useRetryAfter ? getRetryAfterMillis(resDuplicator.duplicateStream())
-                                                     : -1;
+                                       useRetryAfter ?
+                                       getRetryAfterMillis(contentPreviewResponse(resDuplicator)) : -1;
```

Comment:
It seems like you are struggling with indentation rules. :-) How about extracting this into a class so we don't have to worry about indentation? Alternatively, you could use `if`.